bin/assigner/src/main.cpp

```diff
@@ -566,23 +636,21 @@ int curve_dependent_main(std::string bytecode_file_name,
               (target_prover < assigner_instance.assignments.size() || target_prover == invalid_target_prover)) {
         std::uint32_t start_idx = (target_prover == invalid_target_prover) ? 0 : target_prover;
         std::uint32_t end_idx = (target_prover == invalid_target_prover) ? assigner_instance.assignments.size() : target_prover + 1;
+
+        std::vector<std::thread> threads = {};
```

Comment:
May be make sense to use `std::future` instead `threads` (https://en.cppreference.com/w/cpp/thread/future). Automatically join when go out of the local scope